src/cls-ah.ts

```diff
@@ -97,10 +98,11 @@ class AsyncHooksNamespace implements CLSNamespace {
     EVENT_EMITTER_METHODS.forEach((method) => {
       // TODO(kjin): Presumably also dependent on MS/TS-#15473.
       // tslint:disable:no-any
-      const oldMethod = (ee as any)[method];
-      (ee as any)[method] = function(this: {}, event: string, cb: Func<void>) {
-        return oldMethod.call(this, event, ns.bind(cb));
-      };
+      shimmer.wrap(ee, method, (oldMethod) => {
+        return function(this: {}, event: string, cb: Func<void>) {
+          return oldMethod.call(this, event, ns.bind(cb));
```

Comment:
Why is `ns` needed here? Isn't it identical to `this`? In that case, woud
```typescript
return oldMethod.call(this, event, (...args: Array<{}>) => this.call(...args));
```
work?

Reply:
`ns !== this` on line 103, because the value of `this` at line 103 is dynamically scoped.